tests/integration-tests/tests/schedulers/test_slurm.py

```diff
@@ -1557,3 +1621,156 @@ def _test_memory_based_scheduling_enabled_true(
     # check RealMemory overridden via config file parameter
     assert_that(slurm_commands.get_node_attribute("queue1-dy-ondemand1-i3-1", "Memory")).is_equal_to("31400")
     # TODO: Add functional tests for memory-based scheduling
+
+
+def _test_scontrol_reboot_idle_nodes(
+    remote_command_executor,
+    slurm_commands,
+):
+    """Test scontrol reboot with idle nodes."""
+
+    # Get nodes and check that they are idle
+    nodes_in_queue = slurm_commands.get_compute_nodes("queue1")
+    assert_compute_node_states(slurm_commands, nodes_in_queue, "idle")
+
+    # Reboot nodes according to this logic:
+    # - 1 static idle node
+    # - 1 static idle node asap
+    # - 1 dynamic idle node
+    # - 1 dynamic idle node asap
+    static_nodes, dynamic_nodes = get_partition_nodes(nodes_in_queue)
+    assert_that(len(static_nodes)).is_equal_to(2)
+    assert_that(len(dynamic_nodes)).is_equal_to(2)
+    slurm_commands.reboot_compute_node(static_nodes[0], False)
+    slurm_commands.reboot_compute_node(static_nodes[1], True)
+    slurm_commands.reboot_compute_node(dynamic_nodes[0], False)
+    slurm_commands.reboot_compute_node(dynamic_nodes[1], True)
+
+    # Check that nodes enter a reboot state
+    time.sleep(2)
+    assert_compute_node_states(slurm_commands, nodes_in_queue, "reboot^")
+
+    # Wait that nodes come back after a while, without having triggered clustermgtd
+    wait_for_compute_nodes_states(slurm_commands, nodes_in_queue, expected_states=["idle"])
+    assert_no_msg_in_logs(
+        remote_command_executor,
+        ["/var/log/parallelcluster/clustermgtd"],
+        ["Found the following unhealthy static nodes"],
+    )
+
+
+def _test_scontrol_reboot_alloc_nodes(
```

Comment:
The logic in this function seems very similar to the logic in `_test_scontrol_reboot_idle_nodes()`.
Do you think it could make sense to generalize i a single function?

Reply:
I was afraid generalizing it would have made the test workflow less readable, but indeed it came out fine. Addressed in the next commit.